komga/src/main/kotlin/org/gotson/komga/domain/model/ThumbnailSeries.kt

```diff
@@ -6,12 +6,46 @@ import java.net.URL
 import java.time.LocalDateTime
 
 data class ThumbnailSeries(
-  val url: URL,
+  val url: URL?,
   val selected: Boolean = false,
+  var thumbnail: ByteArray?,
 
   val id: String = TsidCreator.getTsid256().toString(),
   val seriesId: String = "",
 
   override val createdDate: LocalDateTime = LocalDateTime.now(),
   override val lastModifiedDate: LocalDateTime = LocalDateTime.now()
-) : Auditable(), Serializable
+) : Auditable(), Serializable {
+
+  val isCustomCover
```

Comment:
The naming is not very explicit. Custom can mean a lot of things. Having a type instead is probably more explicit, and extendable in the future in case there are other types of thumbnails.